lib/dnsmsg-parser/src/ede.rs

```diff
@@ -0,0 +1,102 @@
+use hickory_proto::{
+    error::ProtoResult,
+    serialize::binary::{BinDecodable, BinDecoder, BinEncodable, BinEncoder},
+};
+
+pub const EDE_OPTION_CODE: u16 = 15u16;
+
+#[derive(Debug, Clone)]
+pub struct EDE {
+    info_code: u16,
+    extra_text: Option<String>,
+}
+
+impl EDE {
+    pub fn new(info_code: u16, extra_text: Option<String>) -> Self {
+        Self {
+            info_code,
+            extra_text,
+        }
+    }
+
+    /// Returns the length in bytes of the EdnsOption
+    pub fn len(&self) -> u16 {
+        2 + self.extra_text.as_ref().map(|s| s.len()).unwrap_or(0) as u16
+    }
+
+    pub fn is_empty(&self) -> bool {
+        false
+    }
```

Comment:
Why is this always false?

Reply:
I implemented this for the `len_without_is_empty` clippy warning. Since it refers to the value that `len` returns, this will never be empty, since `len()` always returns a number higher than 2.

Reply:
Ah I see, do we need this function at all then? I see it is used below:

https://github.com/vectordotdev/vector/pull/19937/files#diff-afec58b6de6ae62b9ec7e1c2edcf7ba75c73cac82c6d18bd1a4a0f80b5c8cc3eR90-R96

Maybe we don't need that check if it is always `false`?

Reply:
You are right. I don't think we actually need either `len` or `is_empty`. I have implemented `len` since I followed other types from `hickory_proto`, but I guess that is used for something else inside of that crate. I will just remove both.